lib/spack/spack/url.py

```diff
@@ -199,6 +201,10 @@ def parse_version_offset(path):
         # https://github.com/hpc/mpileaks/releases/download/v1.0/mpileaks-1.0.tar.gz
         (r'github.com/[^/]+/[^/]+/releases/download/v?([^/]+)/.*$', path),
 
+        # e.g., https://gitlab.kitware.com/vtk/vtk/repository/archive.tar.bz2?ref=v7.0.0
+        # e.g., https://gitlab.kitware.com/icet/icet/repository/archive.tar.bz2?ref=IceT-2.1.1
```

Comment:
after `ref=` follows an arbitrary git tag that as usual differs heavily from project to project :)
`IceT` does the weirdness to put it's own name in the git tag for releases.